### PR TITLE
Lucian Q, Gangplank Q, W and Blitz Q coded. Ezreal Q formula. 

### PR DESCRIPTION
**Pull request by DeudlyYT**
- Lucian Q coded
- Gangplank Q and W coded
- Blitzcrank Q coded
- Annie Q coded
- Added Ezreal Q formula
- Lux Ult now damages [Error solved]
- Updated Lucian W, R and Passive structures and Gangplank R and Passive

### CODE REVIEW

File: Champions/Gangplank/Q.lua [hunk 1]
```diff
@@ -1,31 +1,37 @@
...
 
-function finishCasting()
-    local castTarget = castTarget
-    local current = Vector2:new(owner.X, owner.Y)
-    if current:distance(Vector2:new(castTarget.X, castTarget.Y)) <= 625 then    
-        addProjectileTargetCustom("pirate_parley_tar.troy", 0, castTarget)
-    else
-        print("Target is too far away")
-    end
+function onStartCasting()
```
Comment (FurkanS1821) [inline Champions/Gangplank/Q.lua]: Add the projectile stuff under `onFinishCasting()`

File: Champions/Gangplank/Q.lua [hunk 2]
```diff
@@ -1,31 +1,37 @@
...
+	local current = Vector2:new(owner.X, owner.Y)
```
Comment (FurkanS1821) [inline Champions/Gangplank/Q.lua]: These vectors are redundant.

File: Champions/Gangplank/Q.lua [hunk 3]
```diff
@@ -1,31 +1,37 @@
...
-        local owner = owner
-        local damage = getEffectValue(0) + owner:getStats():getTotalAd()
-        local newGold = owner:getStats():getGold() + 3 + 1*spellLevel
-        
-        if castTarget:getStats():getCurrentHealth() >= damage then
-            owner:dealDamageTo(castTarget, damage, DAMAGE_TYPE_PHYSICAL, DAMAGE_SOURCE_SPELL)
+function applyEffects()
+    local target = castTarget
+    local damage = owner:GetStats().AttackDamage.Total + -5 + (25*spellLevel)
+	local newGold = owner:GetStats().Gold + 3 + (1*spellLevel)
```
Comment (FurkanS1821) [inline Champions/Gangplank/Q.lua]: Use space, not tabs.

**Comment by DeudlyYT** [comment]
Is all solved now? @FurkanS1821 @MythicManiac

File: Champions/Ezreal/Q.lua [hunk 4]
```diff
@@ -16,7 +16,7 @@ end
 function applyEffects()
     local AD = owner:GetStats().AttackDamage.Total*1.1
     local AP = owner:GetStats().AbilityPower.Total*0.4
-    local damage = getEffectValue(0) + AD + AP
+	local damage = 15 + (spellLevel*20) + AD + AP
```
Comment (FurkanS1821) [inline Champions/Ezreal/Q.lua]: make this line as how it was (at least it's beginning)

File: Champions/Gangplank/Q.lua [hunk 5]
```diff
@@ -1,31 +1,36 @@
...
-        print("Target is too far away")
-    end
+function onStartCasting()   
     
 end
 
-function applyEffects()
-    local castTarget = castTarget
+function onFinishCasting()
+    local current = Vector2:new(owner.X, owner.Y)
```
Comment (FurkanS1821) [inline Champions/Gangplank/Q.lua]: redundant

File: Champions/Gangplank/Q.lua [hunk 6]
```diff
@@ -1,31 +1,36 @@
...
+    local to = (Vector2:new(spell.X, spell.Y) - current):normalize()
```
Comment (FurkanS1821) [inline Champions/Gangplank/Q.lua]: redundant

File: Champions/Gangplank/Q.lua [hunk 7]
```diff
@@ -1,31 +1,36 @@
...
+    local range = to * 625
```
Comment (FurkanS1821) [inline Champions/Gangplank/Q.lua]: redundant

File: Champions/Gangplank/Q.lua [hunk 8]
```diff
@@ -1,31 +1,36 @@
...
+    local trueCoords = current + range
```
Comment (FurkanS1821) [inline Champions/Gangplank/Q.lua]: redundant

File: Champions/Gangplank/Q.lua [hunk 9]
```diff
@@ -1,31 +1,36 @@
...
+    local target = castTarget
+	
+	addProjectileTarget("pirate_parley_mis", target, false)
```
Comment (FurkanS1821) [inline Champions/Gangplank/Q.lua]: space please

File: Champions/Gangplank/W.lua [hunk 10]
```diff
@@ -1,15 +1,23 @@
...
-    local newHealth = owner:getStats():getCurrentHealth() + 10 + 70*spellLevel + 1*owner:getStats():getTotalAp()
-    local maxHealth = owner:getStats():getMaxHealth()
-    
+function onStartCasting()
+    local newHealth = owner:GetStats().CurrentHealth + 10 + (70*spellLevel) + 1*owner:GetStats().AbilityPower.Total
+    local maxHealth = owner:GetStats().HealthPoints.Total
     
     if newHealth >= maxHealth then
-        owner:getStats():setCurrentHealth(maxHealth)
+		owner:GetStats().CurrentHealth = maxHealth
```
Comment (FurkanS1821) [inline Champions/Gangplank/W.lua]: space please

File: Champions/Gangplank/W.lua [hunk 11]
```diff
@@ -1,15 +1,23 @@
...
     else
-        owner:getStats():setCurrentHealth(newHealth)
+		owner:GetStats().CurrentHealth = newHealth
```
Comment (FurkanS1821) [inline Champions/Gangplank/W.lua]: space please

File: Champions/Gangplank/W.lua [hunk 12]
```diff
@@ -1,15 +1,23 @@
...
     end    
+	addBuff("PirateScurvy", 1, 1, owner, owner)
```
Comment (FurkanS1821) [inline Champions/Gangplank/W.lua]: space please, buffs are not working by the way, is it needed?

File: Champions/Lucian/Q.lua [hunk 13]
```diff
@@ -1,12 +1,27 @@
 Vector2 = require 'Vector2' -- include 2d vector lib 
 
-function onFinishCasting()
-
-	-- THIS IS A LASER SPELL TYPE BUT THE API DOESNT HAVE LASERS
+function onStartCasting()
+	local current = Vector2:new(owner.X, owner.Y)
```
Comment (FurkanS1821) [inline Champions/Lucian/Q.lua]: space please

File: Champions/Lucian/Q.lua [hunk 14]
```diff
@@ -1,12 +1,27 @@
...
+    local to = (Vector2:new(spell.X, spell.Y) - current):normalize()
+    local range = to * 1100
+    local trueCoords = current + range
 	
-	-- SO THIS SPELL CANNOT BE CREATED BY THE MOMENT
+	addLaser(trueCoords.x, trueCoords.y, true)
```
Comment (FurkanS1821) [inline Champions/Lucian/Q.lua]: space please (x5)

File: Champions/Lucian/Q.lua [hunk 15]
```diff
@@ -1,12 +1,27 @@
...
+	addParticleTarget(owner, "Lucian_Q_tar.troy", owner)
+
+end
+function onFinishCasting()
 
 end
-function applyEffects()
 
+function applyEffects()
+	local damage = (owner:GetStats().AttackDamage.Total*(0.45 + (spellLevel*0.15))) + (50 + (spellLevel*30))
```
Comment (FurkanS1821) [inline Champions/Lucian/Q.lua]: space please (x2)

File: Champions/Lucian/W.lua [hunk 16]
```diff
@@ -1,14 +1,23 @@
 Vector2 = require 'Vector2' -- include 2d vector lib 
 
-function onFinishCasting()
+function onStartCasting()
 	local current = Vector2:new(owner.X, owner.Y)
```
Comment (FurkanS1821) [inline Champions/Lucian/W.lua]: space please (x5)

File: Champions/Lucian/W.lua [hunk 17]
```diff
@@ -1,14 +1,23 @@
...
+	
 	addProjectile("LucianWMissile", trueCoords.x, trueCoords.y)
 end
+
+function onFinishCasting()
+
+end
+
 function applyEffects()
 	local damage = 20+spellLevel*40+owner:GetStats().AbilityPower.Total*0.9
```
Comment (FurkanS1821) [inline Champions/Lucian/W.lua]: space please

File: Champions/Lucian/Q.lua [hunk 18]
```diff
@@ -1,12 +1,27 @@
...
 
+function applyEffects()
+    local damage = (owner:GetStats().AttackDamage.Total*(0.45 + (spellLevel*0.15))) + (50 + (spellLevel*30))
+	dealPhysicalDamage(damage)
+	
 end
+
+function onUpdate(diff)
+
+end
```
Comment (MythicManiac) [inline Champions/Lucian/Q.lua]: Missing new line at the end of the file

File: Champions/Lucian/W.lua [hunk 19]
```diff
@@ -1,14 +1,23 @@
...
+
 function applyEffects()
-	local damage = 20+spellLevel*40+owner:GetStats().AbilityPower.Total*0.9
+    local damage = 20+spellLevel*40+owner:GetStats().AbilityPower.Total*0.9
     dealMagicalDamage(damage)
+end
+
+function onUpdate(diff)
+
 end
```
Comment (MythicManiac) [inline Champions/Lucian/W.lua]: Missing new line at the end of the file

File: Champions/Lucian/R.lua [hunk 20]
```diff
@@ -1,7 +1,17 @@
...
+
 end
 
 function applyEffects()
+
+end
+
+function onUpdate(diff)
+
 end
```
Comment (MythicManiac) [inline Champions/Lucian/R.lua]: Missing new line at the end of the file

**Comment by MythicManiac** [comment]
I'd like to see this squashed and the newlines addressed. Overall besides that looking pretty good :+1:

File: Champions/Gangplank/Passive.lua [hunk 21]
```diff
@@ -1,8 +1,25 @@
+Vector2 = require 'Vector2' -- include 2d vector lib
+
 function onUpdate(diff)
+
+end
+
+function onDamageTaken(attacker, damage, type, source)
+    modifyIncomingDamage(newDamage)
```
Comment (MythicManiac) [inline Champions/Gangplank/Passive.lua]: I'd say delete it alltogether

**Pull request closed without merging**